src/components/settings/SettingsLMSTab/tests/CanvasConfig.test.jsx

```diff
@@ -148,24 +181,115 @@ describe('<CanvasConfig />', () => {
+        target: { value: '3' },
+      });
+      fireEvent.change(screen.getByLabelText('Display Name'), {
+        target: { value: 'displayName' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Base URL'), {
+        target: { value: 'https://www.test4.com' },
+      });
     });
-    fireEvent.click(screen.getByText('Cancel'));
-    fireEvent.click(screen.getByText('Save'));
+    expect(screen.getByText('Authorize')).not.toBeDisabled();
+    await act(async () => { userEvent.click(screen.getByText('Authorize')); });
+    await act(async () => { userEvent.click(screen.getByText('Cancel')); });
+    await act(async () => { userEvent.click(screen.getByText('Save')); });
+
     const expectedConfig = {
       active: false,
       display_name: 'displayName',
       enterprise_customer: enterpriseId,
+      canvas_account_id: '3',
+      canvas_base_url: 'https://www.test4.com',
+      client_id: 'test1',
+      client_secret: 'test2',
     };
     expect(LmsApiService.postNewCanvasConfig).toHaveBeenCalledWith(expectedConfig);
   });
+  test('Authorizing a config will initial backend polling', async () => {
+    render(
+      <CanvasConfig
+        enterpriseCustomerUuid={enterpriseId}
+        onClick={mockOnClick}
+        existingData={noExistingData}
+      />,
+    );
+    await act(() => {
+      fireEvent.change(screen.getByLabelText('Display Name'), {
+        target: { value: 'displayName' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Base URL'), {
+        target: { value: 'https://www.test.com' },
+      });
+      fireEvent.change(screen.getByLabelText('API Client ID'), {
+        target: { value: 'test1' },
+      });
+      fireEvent.change(screen.getByLabelText('API Client Secret'), {
+        target: { value: 'test2' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Account Number'), {
+        target: { value: '3' },
+      });
+    });
+    expect(screen.getByText('Authorize')).not.toBeDisabled();
+    await act(async () => { userEvent.click(screen.getByText('Authorize')); });
+    await screen.findByText('Display Name');
+    expect(window.open).toHaveBeenCalled();
+    expect(mockFetchSingleConfig).toHaveBeenCalledWith(1);
+  });
+  test('Authorizing an existing, edited config will call update config endpoint', async () => {
+    render(
+      <CanvasConfig
+        enterpriseCustomerUuid={enterpriseId}
+        onClick={mockOnClick}
+        existingData={existingConfigDataNoAuth}
+      />,
+    );
+    act(() => {
+      fireEvent.change(screen.getByLabelText('Display Name'), {
+        target: { value: 'displayName' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Base URL'), {
+        target: { value: 'https://www.test.com' },
+      });
+    });
+    expect(screen.getByText('Authorize')).not.toBeDisabled();
+    await act(async () => { userEvent.click(screen.getByText('Authorize')); });
```

Comment:
For act warning fixes refer to this PR https://github.com/openedx/frontend-app-enterprise-public-catalog/pull/199/files#diff-fd2c79f0d05a4d6d1caeb7bbe4af38a8f9f39c2885d6e212c499f453cd0c2292R318

generally, the click from userEvent already includes act() so we should not need to wrap those in act. Is my understanding. Usually you want to act() on things you await after clicks have been called, for example

I would not stress much over it if this gets tricky but maybe try it one test and see if that idea works!